### PR TITLE
Feature/kandji

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Adds support for Kandji MDM

# Impacts
New provisioner

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
(REQUIRED, but may be optional...)
## New Features
* Added support for Kandji MDM as a provisioner

### CODE REVIEW

**Comment by julsemaan** [comment]
This is not ready, its missing:

- Documentation in our guide
- Proper testing using our own instance (currently using a dev instance of Kandji and faking some data)

**Comment by julsemaan** [comment]
After testing in our instance everything works except the installation of the agent when the device is jailed in the captive portal. Will check with the Kandji folks if this is something we can achieve, otherwise, it will have to be a limitation.

**Comment by julsemaan** [comment]
PR is ready for review

File: lib/pf/provisioner/kandji.pm [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+as published by the Free Software Foundation; either version 2
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
```
Comment (jrouzierinverse) [inline lib/pf/provisioner/kandji.pm:143]: Should we support polling?

Comment (julsemaan) [reply lib/pf/provisioner/kandji.pm:143]: not in my opinion, most of our provisioners don't and we can always add it later if it becomes a customer requirement.

**Pull request merged**